pkg/server/common/cluster.go

```diff
@@ -0,0 +1,253 @@
+package common
+
+import (
+	"github.com/caicloud/nirvana/log"
+	core_v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/retry"
+
+	api "github.com/caicloud/cyclone/pkg/server/apis/v1alpha1"
+)
+
+// NewClusterClient creates a client for k8s cluster
+func NewClusterClient(c *api.ClusterCredential, inCluster bool) (*kubernetes.Clientset, error) {
+	if inCluster {
+		return newInclusterK8sClient()
+	}
+	return newK8sClient(c)
+}
+
+func newK8sClient(c *api.ClusterCredential) (*kubernetes.Clientset, error) {
+	var config *rest.Config
+	var err error
+
+	// if KubeConfig is not empty, use it firstly, otherwise, use username/password.
+	if c.KubeConfig != nil {
+		config, err = clientcmd.NewDefaultClientConfig(*c.KubeConfig, &clientcmd.ConfigOverrides{}).ClientConfig()
+		if err != nil {
+			log.Infof("NewDefaultClientConfig error: %v", err)
+			return nil, err
+		}
+	} else {
+		if c.TLSClientConfig == nil {
+			c.TLSClientConfig = &api.TLSClientConfig{Insecure: true}
+		}
+
+		config = &rest.Config{
+			Host:        c.Server,
+			BearerToken: c.BearerToken,
+			Username:    c.User,
+			Password:    c.Password,
+			TLSClientConfig: rest.TLSClientConfig{
+				Insecure: c.TLSClientConfig.Insecure,
+				CAFile:   c.TLSClientConfig.CAFile,
+				CAData:   c.TLSClientConfig.CAData,
+			},
+		}
+	}
+
+	client, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		log.Error(err)
+		return nil, err
+	}
+
+	return client, nil
+}
+
+func newInclusterK8sClient() (*kubernetes.Clientset, error) {
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		log.Error(err)
+		return nil, err
+	}
+
+	client, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		log.Error(err)
+		return nil, err
+	}
+
+	return client, nil
+}
+
+// CreateNamespace creates a namespace
+func CreateNamespace(name string, client *kubernetes.Clientset) error {
+	namespace, err := buildNamespace(name)
+	if err != nil {
+		log.Warningf("Build namespace %s error %v", name, err)
+		return err
+	}
+
+	_, err = client.CoreV1().Namespaces().Create(namespace)
+	if err != nil {
```

Comment:
```suggestion
	if _, err = client.CoreV1().Namespaces().Create(namespace); err != nil {
```

Reply:
Do we need to ignore the namespace already exist error ?